core/src/test/java/com/linecorp/armeria/client/retry/RetryingHttpClientTest.java

```diff
@@ -0,0 +1,241 @@
+                        throws Exception {
+                    if (reqCount.getAndIncrement() < 2) {
+                        res.respond(HttpStatus.SERVICE_UNAVAILABLE);
+                    } else {
+                        res.respond(HttpStatus.OK, MediaType.PLAIN_TEXT_UTF_8, "Succeeded after retry");
+                    }
+                }
+            });
+
+            sb.service("/retry-after", new AbstractHttpService() {
+                final AtomicInteger reqCount = new AtomicInteger();
+                @Override
+                protected void doGet(ServiceRequestContext ctx, HttpRequest req, HttpResponseWriter res)
+                        throws Exception {
+                    if (reqCount.getAndIncrement() < 1) {
+                        res.write(HttpHeaders.of(HttpStatus.SERVICE_UNAVAILABLE)
+                                             .set(HttpHeaderNames.RETRY_AFTER, "3"));
+                        res.close();
+                    } else {
+                        res.respond(HttpStatus.OK, MediaType.PLAIN_TEXT_UTF_8, "Succeeded after retry");
+                    }
+                }
+            });
+
+            sb.service("/retry-after-with-rfc-1123", new AbstractHttpService() {
+                final AtomicInteger reqCount = new AtomicInteger();
+                @Override
+                protected void doGet(ServiceRequestContext ctx, HttpRequest req, HttpResponseWriter res)
+                        throws Exception {
+                    if (reqCount.getAndIncrement() < 1) {
+                        final ZonedDateTime after3Sec = ZonedDateTime.now(ZoneOffset.UTC).plusSeconds(3);
+                        final String httpDate = DateTimeFormatter.RFC_1123_DATE_TIME.format(after3Sec);
+                        res.write(HttpHeaders.of(HttpStatus.SERVICE_UNAVAILABLE)
+                                             .set(HttpHeaderNames.RETRY_AFTER, httpDate));
+                        res.close();
+                    } else {
+                        res.respond(HttpStatus.OK, MediaType.PLAIN_TEXT_UTF_8, "Succeeded after retry");
+                    }
+                }
+            });
+
+            sb.service("/retry-after-one-year", new AbstractHttpService() {
+                final AtomicInteger reqCount = new AtomicInteger();
+                @Override
+                protected void doGet(ServiceRequestContext ctx, HttpRequest req, HttpResponseWriter res)
+                        throws Exception {
+                    if (reqCount.getAndIncrement() < 1) {
+                        final ZonedDateTime afterAYear = ZonedDateTime.now(ZoneOffset.UTC).plusYears(1);
+                        final String httpDate = DateTimeFormatter.RFC_1123_DATE_TIME.format(afterAYear);
+                        res.write(HttpHeaders.of(HttpStatus.SERVICE_UNAVAILABLE)
+                                             .set(HttpHeaderNames.RETRY_AFTER, httpDate));
```

Comment:
I'm curious if this is different from `.setTimeMillis(RETRY_AFTER, System.currentTimeMillis() + Duration.ofDays(365).toMillis())`